sdk/src/crypto/raw_signature/rust_native/signers/ecdsa_signer.rs

```diff
@@ -143,6 +156,83 @@ impl TimeStampProvider for EcdsaSigner {
     }
 }
 
+#[derive(DerSequence)]
+struct ECPrivateKey<'a> {
+    version: u32,
+    private_key: &'a [u8], // OCTET STRING content
+    parameters: Option<Any<'a>>,
+    public_key: Option<BitString<'a>>,
+}
+
+fn es512_from_pkcs8_pem(private_key_pem: &str) -> Result<P512SigningKey, RawSignerError> {
+    let pem = pem::parse(private_key_pem).map_err(|e| {
+        RawSignerError::InvalidSigningCredentials(format!("invalid ES512 private key PEM: {e}"))
+    })?;
+    let pk_info = PrivateKeyInfo::try_from(pem.contents()).map_err(|e| {
+        RawSignerError::InvalidSigningCredentials(format!("invalid ES512 PKCS#8 structure: {e}"))
+    })?;
+
+    // Check OID is id-ecPublicKey (1.2.840.10045.2.1)
+    if pk_info.algorithm.oid.as_bytes() != EC_PUBLICKEY_OID.as_bytes() {
+        return Err(RawSignerError::InvalidSigningCredentials(format!(
+            "Unexpected OID: {}, expected id-ecPublicKey {}",
+            pk_info.algorithm.oid, EC_PUBLICKEY_OID
+        )));
+    }
+
+    // Check that algorithm.parameters contains the P-521 OID (1.3.132.0.35)
+    let params = pk_info
+        .algorithm
+        .parameters
+        .as_ref()
+        .ok_or_else(|| {
+            RawSignerError::InvalidSigningCredentials("Missing algorithm parameters".to_string())
+        })?
+        .to_der()
+        .map_err(|_| {
+            RawSignerError::InvalidSigningCredentials(
+                "Algorithm parameters are not a valid OID".to_string(),
+            )
+        })?;
+    // Parse the parameters as an ASN.1 OID
+    let curve_oid = ObjectIdentifier::from_der(&params).map_err(|_| {
+        RawSignerError::InvalidSigningCredentials(
+            "Algorithm parameters are not a valid OID".to_string(),
+        )
+    })?;
+
+    // Now compare the parsed OID to your constant
+    if curve_oid.as_bytes() != SECP521R1_OID.as_bytes() {
+        return Err(RawSignerError::InvalidSigningCredentials(
+            "Unexpected curve OID in parameters: expected P-521 (1.3.132.0.35)".to_string(),
+        ));
+    }
+    // Parse ECPrivateKey ASN.1 structure using asn1-rs
+    let ec_private_key = ECPrivateKey::from_der(pk_info.private_key).map_err(|e| {
+        RawSignerError::InvalidSigningCredentials(format!("invalid ES512 ECPrivateKey ASN.1: {e}"))
+    })?;
+
+    // Verify the private key scalar is 66 bytes unpadded
+    if ec_private_key.1.private_key.len() != 66 {
```

Comment:
Same as above.   Plus this check is wrong since the length is based on the curve.